src/promptflow/promptflow/batch/_base_executor_proxy.py

```diff
@@ -77,10 +78,25 @@ async def exec_line_async(
         inputs: Mapping[str, Any],
         index: Optional[int] = None,
         run_id: Optional[str] = None,
+        enable_stream_output=False,
     ) -> LineResult:
         """Execute a line"""
         raise NotImplementedError()
 
+    def exec_line(
+        self,
+        inputs: Mapping[str, Any],
+        index: Optional[int] = None,
+        run_id: Optional[str] = None,
+        enable_stream_output=False,
+    ) -> LineResult:
```

Comment:
Why add this function in AbstractExecutorProxy?

Reply:
1. I think it is reasonable for executor to also provide a sync interface to execute line, instead of calling async function with `async_run_allowing_running_loop` everywhere. And the implementation (redirect to `exec_line_async`) is genernal